libc/CMakeLists.txt

```diff
@@ -72,6 +72,7 @@ option(LIBC_BUILD_GPU_LOADER "Always build the GPU loader utilities" OFF)
 if(LIBC_BUILD_GPU_LOADER OR (LLVM_LIBC_GPU_BUILD AND NOT LLVM_RUNTIMES_BUILD))
   add_subdirectory(utils/gpu)
 endif()
+option(LIBC_USE_NEW_HEADER_GEN "Generate header files using new headergen instead of the old one" ON)
```

Comment:
for now let's make the default `OFF` so that we can land this, then test it on the appropriate systems, then finally change the default.

Reply:
Understood, I will change this on Aaryan's branch